app/models/service_template_ansible_playbook.rb

```diff
@@ -190,4 +199,20 @@ def delete_job_templates(job_templates)
         .delete_in_provider_queue(job_template.manager.id, { :manager_ref => job_template.manager_ref }, auth_user)
     end
   end
+
+  def extract_single_config(info, action)
+    {}.tap do |hash|
+      [:provision, :retirement, :reconfigure].each do |dummy_action|
+        hash[dummy_action] = {}
+      end
+    end.merge!(action => info)
```

Comment:
Instead of setting up each hash key could you use `Hash.new { |h, k| h[k] = {} }` which would initialize the key to an empty hash one first use?